rclcpp/src/rclcpp/publisher_base.cpp

```diff
@@ -214,7 +214,17 @@ PublisherBase::get_actual_qos() const
     rcl_reset_error();
     throw std::runtime_error(msg);
   }
-  return *qos;
+
+  class RMWQoSIntializedQoS : public rclcpp::QoS
+  {
+  public:
+    explicit
+    RMWQoSIntializedQoS(const rmw_qos_profile_t & rmw_qos):
+    QoS(rclcpp::QoSInitialization::from_rmw(rmw_qos), rmw_qos)
+    {}
+  };
+
+  return RMWQoSIntializedQoS(*qos);
```

Comment:
This subclass looks intentional; why is it necessary? Would this have worked too?

```cpp
return rclcpp::QoS(rclcpp::QoSInitialization::from_rmw(*qos), *qos)
```

Reply:
Oh, for some reason I thought that constructor was protected. I don't need the subclass. I'll update it. Thanks :)

Reply:
863ead4